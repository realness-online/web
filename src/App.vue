<style src="@/style/index.styl" lang="stylus"></style>
<template lang="html">
  <main id="realness">
    <h6>Version {{version}}</h6>
    <router-view></router-view>
    <developer-tools></developer-tools>
  </main>
</template>
<script>
  import developer_tools from '@/components/developer'
  import * as firebase from 'firebase/app'
  export default {
    watch: {
      '$route' (to, from) {
        sessionStorage.previous = from.path
      }
    },
    data() {
      return {
        version: process.env.VUE_APP_VERSION
      }
    },
    created() {
      console.log('initialize app')
      let config = {
        apiKey: process.env.VUE_APP_API_KEY,
        authDomain: process.env.VUE_APP_AUTH_DOMAIN,
        databaseUrl: process.env.VUE_APP_DATABASE_URL,
        projectId: process.env.VUE_APP_PROJECT_ID,
        storageBucket: process.env.VUE_APP_STORAGE_BUCKET,
        messagingSenderId: process.env.VUE_APP_MESSAGING_SENDER_ID
      }
      firebase.initializeApp(config)
    },
    components: {
      'developer-tools': developer_tools
    }
  }
</script>
<style lang="stylus">
  @require '/style/variables'
  #realness > h6
    position:fixed
    bottom: base-line
    left: base-line
</style>
