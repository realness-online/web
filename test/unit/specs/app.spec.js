import Vue from 'vue'
import { shallow } from 'vue-test-utils'
import App from '@/App'

describe('App.vue', () => {
  let vm

  beforeEach(() => {
    vm = shallow(App)
  })

  it('should accept input for posts', () => {
    expect(vm.element).toMatchSnapshot()
  })

  it('should contain a social network', () => {
    expect(vm.element.querySelector('#social_network')).toBeTruthy()
  })

  // it('should list posts')

})
