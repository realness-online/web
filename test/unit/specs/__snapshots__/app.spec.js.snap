// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue should accept input for posts 1`] = `
<main
  id="social_network"
>
  <header>
    <textarea
      placeholder="Wat?"
      tabindex="1"
    />
  </header>
   
  <details>
    <summary>
      Posts
    </summary>
     
    <!---->
  </details>
   
  <details>
    <summary>
      Activity
    </summary>
     
    <!---->
  </details>
</main>
`;
